pkg/sql/catalog/lease/descriptor_version_state.go

```diff
@@ -134,7 +154,30 @@ func (s *descriptorVersionState) incRefCountLocked(ctx context.Context, expensiv
 func (s *descriptorVersionState) getExpiration() hlc.Timestamp {
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	return s.mu.expiration
+
+	// Pick the longest expiration between the session and
```

Comment:
Can we utilize a helper in this function and `hasExpiredLocked`? The logic feels tricky enough that having a single definition is worth while.